src/link/MachO/Object.zig

```diff
@@ -99,10 +99,11 @@ pub fn parse(self: *Object, allocator: Allocator, cpu_arch: std.Target.Cpu.Arch)
             },
             .SYMTAB => {
                 const symtab = cmd.cast(macho.symtab_command).?;
-                self.in_symtab = @ptrCast(
-                    [*]const macho.nlist_64,
-                    @alignCast(@alignOf(macho.nlist_64), &self.contents[symtab.symoff]),
-                )[0..symtab.nsyms];
+                // Sadly, SYMTAB may be at an unaligned offset within the object file.
+                self.in_symtab = @alignCast(@alignOf(macho.nlist_64), @ptrCast(
```

Comment:
Sorry I don't know how I missed this before but these `@alignCast` are doing the exact same assertion they were before. This patch effectively does nothing

Reply:
Hmm, I'm surprised as this fixed it for me locally. I'll re-do the tests then.